### ISSUE TITLE
library.xml has incorrect entry

### ISSUE THREAD

**Issue by tim-moody** [openzim/zimwriterfs#75]
Sorry if this is the wrong repo for this issue.

My understanding is that library.xml is a catalog for http://download.kiwix.org/portable/ and
library_zim.xml is a catalog for http://download.kiwix.org/zim/ BUT

https://mirrors.dotsrc.org/kiwix/library/library.xml contains
``
<book id="96bee5d6-bf84-3942-6ba8-71ea951f5038" title="TED talks - Science" description="Ideas worth spreading" language="eng" creator="TED" publisher="Kiwix" favicon="iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAMAAABg3Am1AAAA9lBMVEX////5vb32oKD84uL6zMz2pqb5wcH+8/Pxbm7rLi75v7/0jo7sNjb0jY3++PjwY2P++vr1lpb6yMj3sLDxbGz2pKT84eHwZ2f71NTsOjr3rq71k5PtR0fsOjr+9vb1kpL96+v+9vb0iYntQUH+9fXvWVn5wcHtQUHvWVn2paX6x8f1mpr+9PT0iIj+8/PtQ0PwXl7wXl7vXFz3pqb6y8v1mpr//PzwZmb5wsLxaWn4tbX719fsOjr4tLT2o6PvWFjsPDz+9/f2oKDyd3f//f3tR0f3qqr//v76y8v1m5v+9vb/+/v6ycn4urr2np73r6/71tb//f0Y5dF0AAAAAWJLR0QAiAUdSAAAAAlwSFlzAAAASAAAAEgARslrPgAAANlJREFUSMdjYBgFo2AUIANGJiTAjMxhgUmxsrEjaeDgRAJcyBxuhBQPLx9pGjg5+QVI1MApCNMgJCwsLAISEQUyxEAMcWEIkABpkBSWkpYBicrKIflDHiSiAGQoghhKyJYrA2kVVZCwGk4N6hoQoAnTwKAFEtbGqQHFD2ANOiCuLgka9PBr0DcAAUMjY7gGE5CwKfGeNjMHCVvg1GBpBQHWIA02VrZ29iBRB0fi/AAHTgykaXB2IUmDq5s7ShL38AQCLyDD2xMJ+PhCaD//gEAa5KtRMAqGOgAAl4sn5xCOsgQAAAAldEVYdGRhdGU6Y3JlYXRlADIwMTQtMDMtMjBUMjE6MDE6MTgrMDE6MDDFcYUrAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDE0LTAzLTA5VDA0OjQwOjQwKzAxOjAwZT6MjwAAABl0RVh0U29mdHdhcmUAQWRvYmUgSW1hZ2VSZWFkeXHJZTwAAAAASUVORK5CYII=" faviconMimeType="image/png" date="2018-07-23" url="http://download.kiwix.org/zim/ted/ted_en_science_2018-07.zim.meta4" articleCount="617" mediaCount="1236" size="14085055"/>

whereas  http://download.kiwix.org/portable/ted does not contain this zim.

This is true of other Ted zims as well.

**Comment by tim-moody** [comment]
As a context, my understanding is that 
All zims with internal FT index are in http://download.kiwix.org/zim/
Some recent zims without internal FT index are in http://download.kiwix.org/zim/
All zims with external FT index are in http://download.kiwix.org/portable/

**Comment by tim-moody** [comment]
@mgautierfr and @kelson42 : any comments?

**Comment by kelson42** [comment]
@tim-moody I think you mostly understand how it works. The point is that we are not generating at all any new ZIP file in `/portable/`. My question is "What is exactly wrong in library.xml" (at least one example).

Remark: library.xml will disappear in the next months as we do not provide any new portable ZIP anymore and people can get the same level of service with library_zim.xml. So I would like to know if this is OK for you. If you still need an answer on this URL, we can make a redirect from library.xml -> library_zim.xml... but we wont distribute ZIP files anymore soon.

**Comment by tim-moody** [comment]
Sorry, I didn't tag the code above so it was not showing:

https://mirrors.dotsrc.org/kiwix/library/library.xml contains
``
<book id="96bee5d6-bf84-3942-6ba8-71ea951f5038" title="TED talks - Science" description="Ideas worth spreading" language="eng" creator="TED" publisher="Kiwix" favicon="iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAMAAABg3Am1AAAA9lBMVEX////5vb32oKD84uL6zMz2pqb5wcH+8/Pxbm7rLi75v7/0jo7sNjb0jY3++PjwY2P++vr1lpb6yMj3sLDxbGz2pKT84eHwZ2f71NTsOjr3rq71k5PtR0fsOjr+9vb1kpL96+v+9vb0iYntQUH+9fXvWVn5wcHtQUHvWVn2paX6x8f1mpr+9PT0iIj+8/PtQ0PwXl7wXl7vXFz3pqb6y8v1mpr//PzwZmb5wsLxaWn4tbX719fsOjr4tLT2o6PvWFjsPDz+9/f2oKDyd3f//f3tR0f3qqr//v76y8v1m5v+9vb/+/v6ycn4urr2np73r6/71tb//f0Y5dF0AAAAAWJLR0QAiAUdSAAAAAlwSFlzAAAASAAAAEgARslrPgAAANlJREFUSMdjYBgFo2AUIANGJiTAjMxhgUmxsrEjaeDgRAJcyBxuhBQPLx9pGjg5+QVI1MApCNMgJCwsLAISEQUyxEAMcWEIkABpkBSWkpYBicrKIflDHiSiAGQoghhKyJYrA2kVVZCwGk4N6hoQoAnTwKAFEtbGqQHFD2ANOiCuLgka9PBr0DcAAUMjY7gGE5CwKfGeNjMHCVvg1GBpBQHWIA02VrZ29iBRB0fi/AAHTgykaXB2IUmDq5s7ShL38AQCLyDD2xMJ+PhCaD//gEAa5KtRMAqGOgAAl4sn5xCOsgQAAAAldEVYdGRhdGU6Y3JlYXRlADIwMTQtMDMtMjBUMjE6MDE6MTgrMDE6MDDFcYUrAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDE0LTAzLTA5VDA0OjQwOjQwKzAxOjAwZT6MjwAAABl0RVh0U29mdHdhcmUAQWRvYmUgSW1hZ2VSZWFkeXHJZTwAAAAASUVORK5CYII=" faviconMimeType="image/png" date="2018-07-23" url="http://download.kiwix.org/zim/ted/ted_en_science_2018-07.zim.meta4" articleCount="617" mediaCount="1236" size="14085055"/>
``

whereas  http://download.kiwix.org/portable/ted does not contain this zim.

This is true of other Ted zims as well.

**Comment by tim-moody** [comment]
@kelson42 for me the issue is whether there are any zims that still only have an **external** full text index, rather than an internal one. Those I get from portable because that is where that index is. This used to be true of large wikis such as en_all and es_all and old ones. I think now new zims _mostly_ have internal.

If you remove portable, old zims that had an external index will lose full text search. An important and popular example is 

http://download.kiwix.org/zim/wikipedia/wikipedia_en_for-schools_2013-01.zim

according to library_xml.zim it has no FT index, but the index 

wikipedia_en_for_schools_opt_2013.zim.idx/

is part of 

http://download.kiwix.org/portable/wikipedia/kiwix-0.9+wikipedia_en_for-schools_2013-01.zip

So to summarize, losing library.xml is not a big problem for me, but losing portable means users lose full text search on some zims.

**Comment by tim-moody** [comment]
I should add that as of the 2018-07 builds none of the EN TED zims has an internal FT index,

**Comment by kelson42** [comment]
@tim-moody Thx for the report, a new library.xml without the problem should be created withing an hour.

**Comment by kelson42** [comment]
@tim-moody I have created a request to make a new ZIM file of Wikipedia for school here: https://github.com/openzim/zim-requests/issues/109 Please do so for any old ZIM files without update you are interested in.

**Comment by mgautierfr** [comment]
> we can make a redirect from library.xml -> library_zim.xml

Please do NOT. We have created `library_zim.xml` especially to avoid confusion and do not break existing scripts. If we decide to remove old portable zim archive, we can remove `library.xml` or empty it, but don't replace it with something else.

**Comment by tim-moody** [comment]
Agree completely, please see #84. I prefer that library.xml remain, but be empty.

**Comment by kelson42** [comment]
`library.xml` is kept and is empty.
